base/compiler/optimize.jl

```diff
@@ -557,7 +557,10 @@ function run_passes(
     @pass "slot2reg"  ir = slot2reg(ir, ci, sv)
     # TODO: Domsorting can produce an updated domtree - no need to recompute here
     @pass "compact 1" ir = compact!(ir)
-    @pass "Inlining"  ir = ssa_inlining_pass!(ir, sv.inlining, ci.propagate_inbounds)
+    # don't inline into toplevel code (which is much slower)
+    if isa(sv.linfo.def, Method)
+        @pass "Inlining"  ir = ssa_inlining_pass!(ir, sv.inlining, ci.propagate_inbounds)
```

Comment:
You need this pass to convert dispatch calls into direct invokes, but can set the parameter to false to stop inlining more